ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java

```diff
@@ -65,41 +66,61 @@ public void createP2PNetwork() throws IOException {
     }
   }
 
-  @Test(expected = IllegalArgumentException.class)
+  @Test
   public void createP2PNetwork_NullHost() throws IOException {
     final NetworkingConfiguration config =
         NetworkingConfiguration.create()
             .setDiscovery(DiscoveryConfiguration.create().setBindHost(null));
-    try (final P2PNetwork broken = builder().config(config).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config);
+    assertThatThrownBy(
+            () -> {
+              try (final P2PNetwork ignored = p2pNetworkBuilder.build()) {
+                Assertions.fail("Expected Exception");
+              }
+            })
+        .isInstanceOf(IllegalArgumentException.class);
   }
 
-  @Test(expected = IllegalArgumentException.class)
+  @Test
   public void createP2PNetwork_InvalidHost() throws IOException {
     final NetworkingConfiguration config =
         NetworkingConfiguration.create()
             .setDiscovery(DiscoveryConfiguration.create().setBindHost("fake.fake.fake"));
-    try (final P2PNetwork broken = builder().config(config).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config);
+    assertThatThrownBy(
+            () -> {
+              try (final P2PNetwork ignored = p2pNetworkBuilder.build()) {
+                Assertions.fail("Expected Exception");
+              }
+            })
+        .isInstanceOf(IllegalArgumentException.class);
   }
 
-  @Test(expected = IllegalArgumentException.class)
+  @Test
   public void createP2PNetwork_InvalidPort() throws IOException {
     final NetworkingConfiguration config =
         NetworkingConfiguration.create()
             .setDiscovery(DiscoveryConfiguration.create().setBindPort(-1));
-    try (final P2PNetwork broken = builder().config(config).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config);
+    assertThatThrownBy(
+            () -> {
+              try (final P2PNetwork ignored = p2pNetworkBuilder.build()) {
+                Assertions.fail("Expected Exception");
+              }
+            })
+        .isInstanceOf(IllegalArgumentException.class);
   }
 
-  @Test(expected = NullPointerException.class)
+  @Test
   public void createP2PNetwork_NullKeyPair() throws IOException {
-    try (final P2PNetwork broken = builder().config(config).nodeKey(null).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config).nodeKey(null);
```

Comment:
nodeKey(null) needs to be inside the lambda (failing test)

Reply:
yes, I just realized that I have to remove the `.build()` call from the lambda assertion. Fixing it